File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/RequestPayPrice.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;
import java.util.UUID;

/**
 * RequestPayPrice
 * viewer -> sim
 *
 * <p>Template: {@code RequestPayPrice Low 161 NotTrusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
public class RequestPayPrice extends SLMessage {
    public ObjectData ObjectData_Field;

    /** Block ObjectData, Single. */
    public static class ObjectData {
        public UUID ObjectID; // LLUUID
    }

    public RequestPayPrice() {
        this.zeroCoded = false;
        this.ObjectData_Field = new ObjectData();
    }

    @Override
    public int CalcPayloadSize() {
        return 20;
    }

    @Override
    public void Handle(SLMessageHandler messageHandler) {
        messageHandler.HandleRequestPayPrice(this);
    }

    @Override
    public void PackPayload(ByteBuffer byteBuffer) {
        // Message number: Low 161 (RequestPayPrice).
        byteBuffer.putShort((short) 0xFFFF);
        byteBuffer.put((byte) 0x00);
        byteBuffer.put((byte) 0xA1);
        packUUID(byteBuffer, this.ObjectData_Field.ObjectID);
    }

    @Override
    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.ObjectData_Field.ObjectID = unpackUUID(byteBuffer);
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/RequestPayPrice.kt
package com.lumiyaviewer.lumiya.slproto.messages

import com.lumiyaviewer.lumiya.slproto.SLMessage
import java.nio.ByteBuffer
import java.util.UUID

/**
 * RequestPayPrice
 * viewer -> sim
 *
 * <p>Template: {@code RequestPayPrice Low 161 NotTrusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
open class RequestPayPrice : SLMessage() {
    @JvmField var ObjectData_Field: ObjectData = ObjectData()

    /** Block ObjectData, Single. */
    open class ObjectData {
        @JvmField var ObjectID: UUID? = null
    }

    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int {
        return 20
    }

    override fun Handle(messageHandler: SLMessageHandler) {
        messageHandler.HandleRequestPayPrice(this)
    }

    override fun PackPayload(byteBuffer: ByteBuffer) {
        // Message number: Low 161 (RequestPayPrice).
        byteBuffer.putShort(0xFFFF.toShort())
        byteBuffer.put((0x00).toByte())
        byteBuffer.put((0xA1).toByte())
        packUUID(byteBuffer, ObjectData_Field.ObjectID)
    }

    override fun UnpackPayload(byteBuffer: ByteBuffer) {
        ObjectData_Field.ObjectID = unpackUUID(byteBuffer)
    }
}
